test_helm_install_uninstall.py

```diff
@@ -0,0 +1,140 @@
+        command = "helm ls -n kube-system"
+        command_output = manager.get_command_output(host_name, command)
+        if len(command_output) == 1:
+            print("Installing driver")
+        else:
+            print("Driver is already installed")
+            return
+
+
+        # copy values.yml file to /root dir on master node
+        current_dir  = subprocess.check_output(['sshpass', '-f', 'INSTALL/passwordFile', 'scp', '-r', 'INSTALL/values_3par_1.18.yaml', 'root@{}:/root'.format(host_ip)])
+
+
+        # ssh to master node and execute adding repo command
+        command = "helm repo add hpe https://hpe-storage.github.io/co-deployments"
+        command_output = manager.get_command_output(host_name, command)
+        assert command_output[0] == '"hpe" has been added to your repositories', "Install of HPE repo failed"
+
+        # ssh to master node and execute helm repo update command
+        command = "helm repo update"
+        command_output = manager.get_command_output(host_name, command)
+        assert command_output[1].rfind('Successfully got an update from the "hpe" chart repository') != -1, "Update of HPE repo failed"
+
+        # ssh to master node and execute helm search command for the repo
+        command = "helm search repo hpe-csi-driver"
+        command_output = manager.get_command_output(host_name, command)
+        chart_version = command_output[1].split()[1]
+        app_version = command_output[1].split()[2]
+
+        # ssh to master node and install csi driver
+        command = "helm install hpe-csi hpe/hpe-csi-driver --namespace kube-system -f values_3par_1.yaml"
+        command_output = manager.get_command_output(host_name, command)
+        #assert command_output[1].rfind('Successfully got an update from the "hpe" chart repository') != -1, "CSI driver installation failed"
+
+        # ssh to master node and check csi driver installation
+        command = "helm ls -n kube-system"
+        command_output = manager.get_command_output(host_name, command)
+        assert command_output[1].rfind('deployed') != -1, "CSI driver not deployed"
+        install_app_version = command_output[1].split()[9]
+        installed_chart = command_output[1].split()[8].split("-")[3]
+        assert install_app_version == app_version,"app version mismatch: repo vesion {0}, installed version {1}".format(app_version, install_app_version)
+        assert installed_chart == chart_version, "chart version mismatch: repo vesion {0}, installed version {1}".format(chart_version, installed_chart)
+
+        # Check plugin pods have been deleted
+        pod_obj = manager.hpe_list_pod_objects("kube-system")
+        pod_list = pod_obj.items
+        for item in pod_list:
+            if 'app' in item.metadata.labels:
+                if item.metadata.labels['app']=='hpe-csi-controller' or item.metadata.labels['app']=='hpe-csi-node' or item.metadata.labels['app']=='primera3par-csp':
```

Comment:
Suggested to use polling as previously mentioned.

Reply:
Already using polling using existing 'manager.check_status' method, which polls until pod is in the "Running" state or when timeout is exceeded